app/assets/src/components/visualizations/Histogram.js

```diff
@@ -301,20 +310,38 @@ export default class Histogram {
     const barCenters = [];
 
     for (let i = 0; i < bins.length; i++) {
-      this.svg
-        .append("g")
-        .attr("class", `bar-${i}`)
-        .attr("fill", colors[i])
-        .selectAll("rect")
-        .data(bins[i])
-        .enter()
-        .append("rect")
-        .attr("class", (_, index) => `rect-${index}`)
-        .attr("x", d => x(d.x0) + i * barWidth)
-        .attr("width", d => barWidth)
-        .attr("y", d => y(d.length))
-        .attr("height", d => y(0) - y(d.length))
-        .style("opacity", barOpacity);
+      // if using log scale for x, the histogram bins should be variable width
+      // so the x and width attributes need to be calculated differently
+      // TODO(julie): is there a better way to consolidate this?
+      this.options.xScaleLog
+        ? this.svg
+            .append("g")
+            .attr("class", `bar-${i}`)
+            .attr("fill", colors[i])
+            .selectAll("rect")
+            .data(bins[i])
+            .enter()
+            .append("rect")
+            .attr("class", (_, index) => `rect-${index}`)
+            .attr("x", d => x(d.x0) + i * ((x(d.x1) - x(d.x0)) / 2))
```

Comment:
Ideally, I would be able to do something like modify `getBarWidth` to calculate this if `this.options.xScaleLog` rather than repeating this giant block of code. @MarkAZhang Do you have any ideas on how to potentially clean this up?

Reply:
Looking again it seems you should be able to get away with the same code. You would just have to replace `2` by the number of series... But would be good to test with coverage viz.

Reply:
By number of series, do you mean `this.data.length`?

Reply:
Hey Julie, you can change the "getBarWidth" function to take a bin.

Then you can do

```
getBarWidth = bin => {
  if (this.options.xScaleLog) {
    return (x(bin.x1) - x(bin.x0)) / 2
  }
  
  // previous code here
}
```
 
```
.attr("x", d => x(d.x0) + i * getBarWidth(d))
.attr("width", d => getBarWidth(d))
```

Reply:
> By number of series, do you mean `this.data.length`?

Yes

Reply:
Took Mark's suggestion, thanks! I'll be sure to double check the coverage viz once this is on staging as well to make sure this doesn't affect anything there.